src/main/java/org/opentripplanner/middleware/docs/PublicApiDocGenerator.java

```diff
@@ -0,0 +1,416 @@
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Predicate;
+
+import static org.opentripplanner.middleware.utils.ConfigUtils.getConfigPropertyAsText;
+
+/**
+ * Class that generates the public-facing API documentation.
+ */
+public class PublicApiDocGenerator {
+    private static final Logger LOG = LoggerFactory.getLogger(PublicApiDocGenerator.class);
+    private static final HttpClient client = HttpClient.newBuilder().build();
+    private static final String API_TEMPLATE_FILE = "api-swagger-template.yaml";
+
+    /**
+     * List of public API paths published in the public-facing API documentation.
+     * TODO: List any new public endpoints here in order for them to be published.
+     */
+    private static final String[] PUBLIC_PATHS = new String[] {
+        "api/secure/monitoredtrip",
+        "api/secure/triprequests",
+        "api/secure/user",
+        "otp",
+        "pelias"
+    };
+    private static final String DEFINITIONS_REF = "#/definitions/";
+    private static final String REF_FIELD = "$ref";
+    private static final String ABSTRACT_USER = "AbstractUser";
+    private static final String OTP_USER = "OtpUser";
+    private static final String AUTOGENERATED_DOC_URL = "http://localhost:4567/doc.yaml";
+    // Use the system temporary directory to write the output file
+    private static final String OUTPUT_FILE = System.getProperty("java.io.tmpdir") + File.separator + "otp-middleware-swagger.yaml";
+    private static final String JAR_PREFIX = "otp-middleware-";
+
+    private static final String AWS_API_SERVER = getConfigPropertyAsText("AWS_API_SERVER");
+    private static final String AWS_API_STAGE = getConfigPropertyAsText("AWS_API_STAGE");
+
+    // Cached variables.
+    private final ObjectNode rootNode;
+    private final JsonNode templateJson;
+    private final ObjectNode pathsNode;
+    private final ObjectNode definitionsNode;
+    private final ArrayNode tagsNode;
+
+    public PublicApiDocGenerator() throws IOException, InterruptedException {
+        // Start with the swagger YAML skeleton generated by spark-swagger (clone it)
+        String autoGeneratedSwagger = getSwaggerDocsAsString();
+
+        // The document we are working off and related nodes.
+        rootNode =  YamlUtils.yamlMapper.readTree(autoGeneratedSwagger).deepCopy();
+        definitionsNode = (ObjectNode) rootNode.get("definitions");
+        pathsNode = (ObjectNode) rootNode.get("paths");
+        tagsNode = (ArrayNode) rootNode.get("tags");
+
+        // Load definitions from a template file.
+        templateJson = YamlUtils.yamlMapper.readTree(OtpMiddlewareMain.class.getClassLoader().getResourceAsStream(API_TEMPLATE_FILE));
+    }
+
+    /**
+     * Generate an enhanced public-facing Swagger document from the one auto-generated by spark-swagger.
+     * This handles things like adding security parameters, method responses, and removing restricted API resources
+     * from the public-facing document.
```

Comment:
"Public-facing" should be clarified a bit more. 

```
Generate an enhanced public-facing (i.e., the collection of endpoints available to an {@link ApiUser}, which excludes admin endpoints like viewing application errors or request logs) Swagger document....
```

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/59/commits/b327a799b261096bd67fcaf0d680d92b1090fc2a